### DISCUSSION TITLE
Integrate arduino library

### DISCUSSION THREAD

**Discussion by pabou38** [micropython/micropython#14276] [ESP32]
I am looking for a way to integrate some arduino code into micropython, as a "custom module".

The context is that I want to use tensorflow lite for microcontroller, which is available as an arduino library for the esp32, but much much prefer to write the app in micropython.

I just need two functions , an init (), and a predict(). The arduino code (which I would write) for those two will take care of all the tflite related stuff. 

Tflite  for microcontroller allows to perform machine learning inference directly on the microcontroller. There is no micropython version of this.

Is this doable ? am I looking for way too much trouble?

Thanks for any advice 

Pascal boudalier

**Comment by scruss** [comment]
[mocleiri/tensorflow-micropython-examples: A custom micropython firmware integrating tensorflow lite for microcontrollers and ulab to implement the tensorflow micro examples.](https://github.com/mocleiri/tensorflow-micropython-examples) (which builds on ESP32) is available.
